modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java

```diff
@@ -0,0 +1,176 @@
+import org.testcontainers.utility.MountableFile;
+
+/**
+ * See
+ * <a href="https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0">https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0</a>
+ */
+public class InfluxDBContainerV2<SELF extends InfluxDBContainerV2<SELF>> extends GenericContainer<SELF> {
+
+    public static final Integer INFLUXDB_PORT = 8086;
+
+    private static final String REGISTRY = "quay.io";
+    private static final String REPOSITORY = "influxdb/influxdb";
+    private static final String TAG = "v2.0.0";
+    private static final DockerImageName DEFAULT_IMAGE_NAME =
+        DockerImageName.parse(String.format("%s/%s:%s", REGISTRY, REPOSITORY, TAG));
+    private static final int NO_CONTENT_STATUS_CODE = 204;
+    private static final String INFLUX_SETUP_SH = "influx-setup.sh";
+
+    private String username = "test-user";
+    private String password = "test-password";
+    private String bucket = "test-bucket";
+    private String organization = "test-org";
+    private int retention = 0;
+    private String retentionUnit = RetentionUnit.NANOSECONDS.label;
+
+    private InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.waitStrategy = (new WaitAllStrategy())
+            .withStrategy(Wait
+                .forHttp("/ping")
+                .withBasicCredentials(this.username, this.password)
+                .forStatusCode(NO_CONTENT_STATUS_CODE))
+            .withStrategy(Wait.forListeningPort());
+        this.addExposedPort(INFLUXDB_PORT);
+    }
+
+    public static InfluxDBContainerV2<?> createWithDefaultTag() {
+        return new InfluxDBContainerV2<>(DEFAULT_IMAGE_NAME);
+    }
+
+    public static InfluxDBContainerV2<?> createWithSpecificTag(final DockerImageName imageName) {
+        return new InfluxDBContainerV2<>(imageName);
+    }
+
+    @Override
+    protected void configure() {
+        this.addEnv("INFLUXDB_USER", this.username);
+        this.addEnv("INFLUXDB_PASSWORD", this.password);
+        this.addEnv("INFLUXDB_BUCKET", this.bucket);
+        this.addEnv("INFLUXDB_ORG", this.organization);
+        this.addEnv("INFLUXDB_RETENTION", String.valueOf(this.retention));
+        this.addEnv("INFLUXDB_RETENTION_UNIT", this.retentionUnit);
+    }
+
+    @Override
+    @SneakyThrows({InterruptedException.class, IOException.class, ExecutionException.class})
+    public void start() {
+        this.withCopyFileToContainer(MountableFile.forClasspathResource(INFLUX_SETUP_SH),
+            String.format("%s", INFLUX_SETUP_SH));
+        if (this.containerId != null) {
+            return;
+        }
+        Startables.deepStart(this.dependencies).get();
+        // trigger LazyDockerClient's resolve so that we fail fast here and not in getDockerImageName()
+        this.dockerClient.authConfig();
```

Comment:
I don't quite understand why this is here...

Reply:
I have removed this